GetSubscription/handler.ts

```diff
@@ -0,0 +1,113 @@
+import { Context } from "@azure/functions";
+import * as express from "express";
+import * as TE from "fp-ts/lib/TaskEither";
+import {
+  AzureApiAuthMiddleware,
+  IAzureApiAuthorization,
+  UserGroup
+} from "@pagopa/io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import { ContextMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredParamMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_param";
+import { withRequestMiddlewares } from "@pagopa/io-functions-commons/dist/src/utils/request_middleware";
+import { wrapRequestHandler } from "@pagopa/ts-commons/lib/request_middleware";
+import {
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseSuccessJson,
+  ResponseSuccessJson
+} from "@pagopa/ts-commons/lib/responses";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { pipe } from "fp-ts/lib/function";
+
+import { Subscription } from "../generated/definitions/Subscription";
+import {
+  getApiClient,
+  getSubscription,
+  mapApimRestError,
+  IAzureApimConfig,
+  IServicePrincipalCreds,
+  parseOwnerIdFullPath
+} from "../utils/apim";
+import { genericInternalErrorHandler } from "../utils/errorHandler";
+
+type IGetSubscriptionHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  subscriptionid: NonEmptyString
+) => Promise<
+  | IResponseSuccessJson<Subscription>
+  | IResponseErrorInternal
+  | IResponseErrorNotFound
+>;
+
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+export function GetSubscriptionHandler(
+  servicePrincipalCreds: IServicePrincipalCreds,
+  azureApimConfig: IAzureApimConfig
+): IGetSubscriptionHandler {
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  return async (context, _, subscriptionId) => {
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    const internalErrorHandler = (errorMessage: string, error: Error) =>
+      genericInternalErrorHandler(
+        context,
+        "GetSubscription | " + errorMessage,
+        error,
+        errorMessage
+      );
+    return pipe(
+      getApiClient(servicePrincipalCreds, azureApimConfig.subscriptionId),
+      TE.mapLeft(error =>
+        internalErrorHandler("Could not get the APIM client.", error)
+      ),
+      TE.chain(apim =>
+        pipe(
+          getSubscription(
+            apim,
+            azureApimConfig.apimResourceGroup,
+            azureApimConfig.apim,
+            subscriptionId
+          ),
+          TE.mapLeft(mapApimRestError("Subscription")),
+          TE.map(subscription =>
+            ResponseSuccessJson({
+              id: subscription.id,
+              owner_id: parseOwnerIdFullPath(
+                subscription.ownerId as NonEmptyString
+              ),
+              primary_key: subscription.primaryKey,
```

Comment:
Do we need to expose `primary_key` and `secondary_key` in this API? 
If not striclty required, I'd rather not expose such information. 
If it's required it's still ok, let just spend some time reasoning about how we ensure nobody is getting an apykey of a subscription it doesn't own.

(I see this API is protected at admin level, but for the _least data principle_ we might be mimimize the data we expose.)

Reply:
In our specific case, basically, only the ownerId of the subscription is needed.
But since it is a generic API we decided not to alter the Subscription interface except for the part that not concerning the concept of subscription itself _(i.e. the service)_.
We can decide together whether to restrict it to only the data necessary for our purpose or whether to expand the amount of data to be displayed in continuity with the current **Subscription** interface.